allauth_2fa/adapter.py

```diff
@@ -38,9 +44,34 @@ def get_2fa_authenticate_url(self, request):
             view.request = request
             success_url = view.get_success_url()
             query_params = request.GET.copy()
-            if success_url:
-                query_params[view.redirect_field_name] = success_url
+            next_url = get_next_redirect_url(request)
+            if next_url:
+                query_params["next"] = next_url
             if query_params:
                 redirect_url += f"?{urlencode(query_params)}"
-
         return redirect_url
+
+    def stash_pending_login(self, request, user, **kwargs):
```

Comment:
:heavy_check_mark: Added docstring. As for the copy, a shallow one is fine. A deep one would dupe the whole `sociallogin` model which goes to far -- it is serializer in this method anyway.

Reply:
Just to clarify, the goal is to prevent changes inside the `stash` method from leaking outside. I want to prevent this:

```
data = {...}
stash(data)
# expected result: `data` itself is unchanged.
```

Given that `stash()` is setting new keys, a shallow copy is all that is needed. As, setting new keys does not mutate the values.